backend/fs/BlockFS.cpp

```diff
@@ -250,7 +250,7 @@ bool BlockFS::deleteSource(const std::string &name_or_id) {
 //--------------------------------------------------
 
 std::shared_ptr<base::IDataArray> BlockFS::createDataArray(const std::string &name, const std::string &type,
-                                                           nix::DataType data_type, const NDSize &shape) {
+                                                           nix::DataType data_type, const NDSize &shape, bool compression) {
```

Comment:
Although I indeed very much like the simplicity of the approach, maybe we can be a tiny bit more extensible and future-proof by having an enum instead of the bool, ie:
```c++
enum class Compression {
   None,
   DeflateNormal,

   Auto = DeflateNormal,
   //and later:
   DeflateHigh,
   DeflateLow,

    FileDefault, // use the default store within the file
}
```